zarr/convenience.py

```diff
@@ -73,7 +73,8 @@ def open(store=None, mode='a', **kwargs):
     path = kwargs.get('path', None)
     # handle polymorphic store arg
     clobber = mode == 'w'
-    store = normalize_store_arg(store, clobber=clobber)
+    store = normalize_store_arg(store, clobber=clobber,
+                                storage_options=kwargs.pop("storage_options", {}))
```

Comment:
I would appreciate a comment here so that the other devs understand why were are treating this argument specially.

Reply:
done